examples/llm_plus_gnn/benchmark_model_gnn_enc.py

```diff
@@ -0,0 +1,37 @@
+# %%
+from g_retriever import benchmark_models, get_loss, inference_step
+
+from torch_geometric.datasets import UpdatedWebQSPDataset
+from torch_geometric.nn.models import GAT, MLP, GRetriever
+
+# %%
+hidden_channels = 1024
+num_gnn_layers = 1
+lr=1e-5
+epochs=2
+batch_size=8
+eval_batch_size=16
+
+# %%
+ds = UpdatedWebQSPDataset('benchmark_archs')
+
+# %%
+model_names = []
+model_classes = []
+model_kwargs = []
+model_type = ["GAT"]
+models = {"GAT": GAT, "MLP": MLP}
+num_layers = [4]
+num_tokens = [4, 16]
+for m_type in model_type:
+    for n_layer in num_layers:
+        for n_tokens in num_tokens
+            model_names.append(f"{m_type}_{n_layer}_{n_tokens}")
+            model_classes.append(GRetriever)
+            kwargs = dict(gnn_hidden_channels=hidden_channels*n_tokens, num_gnn_layers=n_layer, gnn_to_use=models[m_type], gnn_out_channels=hidden_channels*n_tokens, mlp_out_tokens=n_tokens)
```

Comment:
whats the major diff btwn `examples/llm_plus_gnn/benchmark_model_gnn_enc.py` and `examples/llm_plus_gnn/benchmark_model_archs.py`? would likely be good to consolidate these.
also in general benchmarks should go in the benchmarks folder of pyg, not examples

Reply:
`gnn_enc` enables increasing the number of tokens that are provided by the GNN, along with increasing the size of the GNN architecture proportionally. We can refactor out of this PR if we want to limit the scope of changes.

Reply:
okay thats fine, just need to add more comments explaining that